File: src/unit-test/bubble_sort.go
package main

import "fmt"

func bubbleSort(arr []int) []int {
	n := len(arr)
	
	for i := 0; i < n-1; i++ {
		for j := 0; j < n-i-1; j++ {
			if arr[j] > arr[j+1] {
				// Swap elements
				arr[j], arr[j+1] = arr[j+1], arr[j]
			}
		}
	}
	
	return arr
}

func main() {
	array := []int{64, 34, 25, 12, 22, 11, 90}
	
	fmt.Println("Original array:", array)
	
	sortedArray := make([]int, len(array))
	copy(sortedArray, array)
	sortedArray = bubbleSort(sortedArray)
	
	fmt.Println("Sorted array:", sortedArray)
}

File: src/unit-test/bubble_sort.cpp
#include <iostream>
#include <vector>

void bubbleSort(std::vector<int>& arr) {
    int n = arr.size();
    
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                // Swap elements
                std::swap(arr[j], arr[j + 1]);
            }
        }
    }
}

void printArray(const std::vector<int>& arr) {
    for (int value : arr) {
        std::cout << value << " ";
    }
    std::cout << std::endl;
}

int main() {
    std::vector<int> arr = {64, 34, 25, 12, 22, 11, 90};
    
    // 输出原始数组
    std::cout << "Original array: " << std::endl;
    printArray(arr);
    
    // 创建原始数组的副本，用于排序
    std::vector<int> sortedArr = arr;
    bubbleSort(sortedArr);
    
    // 输出排序后的数组
    std::cout << "Sorted array: " << std::endl;
    printArray(sortedArr);
    
    return 0;
}